installer/src/uz801.rs

```diff
@@ -0,0 +1,222 @@
+/// Installer for the Uz801 hotspot.
+///
+/// Installation process:
+/// 1. Use curl to activate USB debugging backdoor
+/// 2. Wait for device reboot and ADB availability
+/// 3. Use ADB to install rayhunter files
+/// 4. Modify startup script to launch rayhunter on boot
+use std::time::Duration;
+
+use adb_client::{ADBDeviceExt, ADBUSBDevice, RustADBError};
+use anyhow::Result;
+use std::io::ErrorKind;
+use tokio::time::sleep;
+
+use crate::Uz801Args as Args;
+use crate::util::echo;
+
+pub async fn install(
+    Args {
+        admin_ip,
+    }: Args,
+) -> Result<()> {
+    run_install(admin_ip).await
+}
+
+async fn run_install(admin_ip: String) -> Result<()> {
+    echo!("Activating USB debugging backdoor... ");
+    activate_usb_debug(&admin_ip).await?;
+    println!("ok");
+
+    echo!("Waiting for device reboot and ADB connection... ");
+    let mut adb_device = wait_for_adb().await?;
+    println!("ok");
+
+    echo!("Installing rayhunter files... ");
+    install_rayhunter_files(&mut adb_device).await?;
+    println!("ok");
+
+    echo!("Modifying startup script... ");
+    modify_startup_script(&mut adb_device).await?;
+    println!("ok");
+
+    echo!("Starting rayhunter daemon... ");
+    start_rayhunter(&mut adb_device).await?;
+    println!("ok");
+
+    echo!("Testing rayhunter... ");
+    test_rayhunter(&admin_ip).await?;
+    println!("ok");
+    println!("rayhunter is running at http://{admin_ip}:8080");
+
+    Ok(())
+}
+
+pub async fn activate_usb_debug(admin_ip: &str) -> Result<()> {
+    let url = format!("http://{}/ajax", admin_ip);
+    let referer = format!("http://{}/usbdebug.html", admin_ip);
+    let origin = format!("http://{}", admin_ip);
+    
+    let _handle = tokio::spawn(async move {
+        let client = reqwest::Client::builder()
+            .timeout(Duration::from_secs(5))
+            .build()
+            .unwrap();
+            
+        let _response = client
+            .post(&url)
+            .header("Accept", "application/json, text/javascript, */*; q=0.01")
```

Comment:
I think most of those headers are probably not necessary but you'd have to try it out.

Reply:
I removed some that would definitely be unnecessary, but I did not have a way test activation repeatedly (as you said.) I'll look for what it does and if it is reversible, e.g. some flag, I will report on how to reverse it. It'll be useful for the uninstallation guide in theory.

Reply:
Activation is now reversible; the installer would just have to run `setprop persist.sys.usb.config rndis`.